core/services/helpers/datetime/AbstractHelper.php

```diff
@@ -11,27 +11,84 @@
 
 abstract class AbstractHelper implements HelperInterface
 {
+    /**
+     * @var string
+     */
+    private $default_timezone_string = '';
+
+
+    /**
+     * AbstractHelper constructor.
+     */
+    public function __construct()
+    {
+        add_action('update_option_timezone_string', [$this, 'updateDefaultTimezoneString'], 10, 2);
```

Comment:
IF the timezone is somehow changed during a request, then this resets the cached value here so that the new value gets picked up.